tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['key_format'] == 'PASSPHRASE', str(job_status_result)
+            assert dictionary['unlock_successful'] is False, str(job_status_result)
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_20_unlock_passphrase_encrypted_datasets_and_ensure_they_get_unlocked():
+    payload = {
+        'id': dataset,
+        'unlock_options': {
+            'recursive': True,
+            'datasets': [
+                {
+                    'name': dataset,
+                    'passphrase': 'my_passphrase'
+                }
+            ]
+        }
+    }
+    results = POST('/pool/dataset/unlock/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_21_verify_passphrase_encrypted_root_is_unlocked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['key_format'] == 'PASSPHRASE', str(job_status_result)
+            assert dictionary['unlock_successful'] is True, str(job_status_result)
+            assert dictionary['locked'] is False, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_22_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_23_delete_pool():
+    payload = {
+        'cascade': True,
+        'restart_services': True,
+        'destroy': True
+    }
+    results = POST(f'/pool/id/{pool_id}/export/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_24_create_a_passphrase_encrypted_pool():
+    global pool_id
+    payload = {
+        'name': pool_name,
+        'encryption': True,
+        'encryption_options': {
+            'algorithm': 'AES-128-CCM',
```

Comment:
Let's please assert the `algorithm` to ensure that the specified one is indeed being used